### DISCUSSION TITLE
Vehicle Routing Problem Profits

### DISCUSSION THREAD

**Discussion by rahulj2409** [google/or-tools#2638] [Routing (and legacy CP) questions]
Is there a way to change the problem to each task has a profit assigned to it. 
Furthermore, instead of having to complete all tasks, 
1. you have a limited amount of drivers and the goal is to find the amount of tasks that give maximum profit that can be done among all drivers given the time window constraint 
2. OR, each driver can only do a certain amount of tasks so how can you find the amount of tasks per driver that gives maximum profit overall.

What are the ways to solve both different questions.

**Comment by Mizux** [comment]
1. You may use AddDisjunction() with a penalty proportionate to the profit I;e. better to drop less profitable nodes

2. (1 bis) you could also use the prize collecting SAT example (one driver only):  https://github.com/google/or-tools/blob/stable/examples/python/prize_collecting_tsp_sat.py (this use the CP-SAT solver instead of the routing solver)

3. What do you means per "amount of task" ? if you means number of visit, then you could use a counter/capacity dimension with +1 or -1 at each node visited and use the vehicle capacity to limit the number of max visit allowed...
e.g. https://github.com/google/or-tools/blob/stable/ortools/constraint_solver/samples/vrp_tokens.py

**Reply by rahulj2409** [reply]
Ah that makes sense, thanks!
What i meant is number of visits. How does one use a counter/capacity dimension with +1 or -1 at each node visited? Is there an example of the code here, thanks!

Thanks
